domain/duration.go

```diff
@@ -33,3 +33,7 @@ func (d YearWithSemesterDuration) IsValid() bool {
 
 	return u.After(s) && s.Year >= 1970 && u.Year < 2070 && s.Semester >= 0 && s.Semester < 2 && u.Semester >= 0 && u.Semester < 2
 }
+
+func (out YearWithSemesterDuration) IsIncludes(in YearWithSemesterDuration) bool {
+	return !out.Since.After(in.Since) || in.Until.After(out.Until)
```

Comment:
`out.Since <= in.Since && in.Until <= out.Until`だからこう？自信ないです
```suggestion
	return !out.Since.Before(in.Since) && !in.Until.Before(out.Until)
```